samples/boards/intel_s1000_crb/audio/sample.board.intel_s1000_crb.audio Fails to build

Doing:

`./scripts/sanitycheck -p intel_s1000_crb -s samples/boards/intel_s1000_crb/audio/sample.board.intel_s1000_crb.audio`

Gets:

```
[ 10%] Building C object CMakeFiles/app.dir/src/usb_transport.c.obj
/home/galak/git/zephyr/samples/boards/intel_s1000_crb/audio/src/usb_transport.c:34:16: error: initialization of 'int (*)(const struct device *, struct usb_setup_packet *, int32_t *, uint8_t **)' {aka 'int (*)(const struct device *, struct usb_setup_packet *, int *, unsigned char **)'} from incompatible pointer type 'int (*)(struct usb_setup_packet *, int32_t *, uint8_t **)' {aka 'int (*)(struct usb_setup_packet *, int *, unsigned char **)'} [-Werror=incompatible-pointer-types]
   34 |  .get_report = usb_transport_get_report,
      |                ^~~~~~~~~~~~~~~~~~~~~~~~
/home/galak/git/zephyr/samples/boards/intel_s1000_crb/audio/src/usb_transport.c:34:16: note: (near initialization for 'usb_transport_callbacks.get_report')
/home/galak/git/zephyr/samples/boards/intel_s1000_crb/audio/src/usb_transport.c:35:14: error: initialization of 'int (*)(const struct device *, struct usb_setup_packet *, int32_t *, uint8_t **)' {aka 'int (*)(const struct device *, struct usb_setup_packet *, int *, unsigned char **)'} from incompatible pointer type 'int (*)(struct usb_setup_packet *, int32_t *, uint8_t **)' {aka 'int (*)(struct usb_setup_packet *, int *, unsigned char **)'} [-Werror=incompatible-pointer-types]
   35 |  .get_idle = usb_transport_get_idle,
      |              ^~~~~~~~~~~~~~~~~~~~~~
/home/galak/git/zephyr/samples/boards/intel_s1000_crb/audio/src/usb_transport.c:35:14: note: (near initialization for 'usb_transport_callbacks.get_idle')
/home/galak/git/zephyr/samples/boards/intel_s1000_crb/audio/src/usb_transport.c:36:18: error: initialization of 'int (*)(const struct device *, struct usb_setup_packet *, int32_t *, uint8_t **)' {aka 'int (*)(const struct device *, struct usb_setup_packet *, int *, unsigned char **)'} from incompatible pointer type 'int (*)(struct usb_setup_packet *, int32_t *, uint8_t **)' {aka 'int (*)(struct usb_setup_packet *, int *, unsigned char **)'} [-Werror=incompatible-pointer-types]
   36 |  .get_protocol = usb_transport_get_protocol,
      |                  ^~~~~~~~~~~~~~~~~~~~~~~~~~
/home/galak/git/zephyr/samples/boards/intel_s1000_crb/audio/src/usb_transport.c:36:18: note: (near initialization for 'usb_transport_callbacks.get_protocol')
/home/galak/git/zephyr/samples/boards/intel_s1000_crb/audio/src/usb_transport.c:37:16: error: initialization of 'int (*)(const struct device *, struct usb_setup_packet *, int32_t *, uint8_t **)' {aka 'int (*)(const struct device *, struct usb_setup_packet *, int *, unsigned char **)'} from incompatible pointer type 'int (*)(struct usb_setup_packet *, int32_t *, uint8_t **)' {aka 'int (*)(struct usb_setup_packet *, int *, unsigned char **)'} [-Werror=incompatible-pointer-types]
   37 |  .set_report = usb_transport_set_report,
      |                ^~~~~~~~~~~~~~~~~~~~~~~~
/home/galak/git/zephyr/samples/boards/intel_s1000_crb/audio/src/usb_transport.c:37:16: note: (near initialization for 'usb_transport_callbacks.set_report')
/home/galak/git/zephyr/samples/boards/intel_s1000_crb/audio/src/usb_transport.c:38:14: error: initialization of 'int (*)(const struct device *, struct usb_setup_packet *, int32_t *, uint8_t **)' {aka 'int (*)(const struct device *, struct usb_setup_packet *, int *, unsigned char **)'} from incompatible pointer type 'int (*)(struct usb_setup_packet *, int32_t *, uint8_t **)' {aka 'int (*)(struct usb_setup_packet *, int *, unsigned char **)'} [-Werror=incompatible-pointer-types]
   38 |  .set_idle = usb_transport_set_idle,
      |              ^~~~~~~~~~~~~~~~~~~~~~
/home/galak/git/zephyr/samples/boards/intel_s1000_crb/audio/src/usb_transport.c:38:14: note: (near initialization for 'usb_transport_callbacks.set_idle')
/home/galak/git/zephyr/samples/boards/intel_s1000_crb/audio/src/usb_transport.c:39:18: error: initialization of 'int (*)(const struct device *, struct usb_setup_packet *, int32_t *, uint8_t **)' {aka 'int (*)(const struct device *, struct usb_setup_packet *, int *, unsigned char **)'} from incompatible pointer type 'int (*)(struct usb_setup_packet *, int32_t *, uint8_t **)' {aka 'int (*)(struct usb_setup_packet *, int *, unsigned char **)'} [-Werror=incompatible-pointer-types]
   39 |  .set_protocol = usb_transport_set_protocol,
      |                  ^~~~~~~~~~~~~~~~~~~~~~~~~~
/home/galak/git/zephyr/samples/boards/intel_s1000_crb/audio/src/usb_transport.c:39:18: note: (near initialization for 'usb_transport_callbacks.set_protocol')
/home/galak/git/zephyr/samples/boards/intel_s1000_crb/audio/src/usb_transport.c:40:18: error: initialization of 'void (*)(const struct device *)' from incompatible pointer type 'void (*)(void)' [-Werror=incompatible-pointer-types]
   40 |  .int_in_ready = usb_transport_host_ready,
      |                  ^~~~~~~~~~~~~~~~~~~~~~~~
/home/galak/git/zephyr/samples/boards/intel_s1000_crb/audio/src/usb_transport.c:40:18: note: (near initialization for 'usb_transport_callbacks.int_in_ready')
cc1: all warnings being treated as errors
gmake[2]: *** [CMakeFiles/app.dir/build.make:135: CMakeFiles/app.dir/src/usb_transport.c.obj] Error 1
gmake[1]: *** [CMakeFiles/Makefile2:2306: CMakeFiles/app.dir/all] Error 2
gmake: *** [Makefile:104: all] Error 2
```



INFO    - 5/8 intel_s1000_crb           samples/boards/intel_s1000_crb/audio/sample.board.intel_s1000_crb.audio FAILED Build failure (build)
 

--- a/samples/boards/intel_s1000_crb/audio/src/usb_transport.c
+++ b/samples/boards/intel_s1000_crb/audio/src/usb_transport.c
@@ -16,19 +16,25 @@ LOG_MODULE_REGISTER(usb_transport);
 #include "usb_transport.h"
 
 /* callback function list */
-static int usb_transport_get_report(struct usb_setup_packet *setup,
+static int usb_transport_get_report(const struct device *dev,
+		struct usb_setup_packet *setup,
 		int32_t *len, uint8_t **data);
-static int usb_transport_get_idle(struct usb_setup_packet *setup,
+static int usb_transport_get_idle(const struct device *dev,
+		struct usb_setup_packet *setup,
 		int32_t *len, uint8_t **data);
-static int usb_transport_get_protocol(struct usb_setup_packet *setup,
+static int usb_transport_get_protocol(const struct device *dev,
+		struct usb_setup_packet *setup,
 		int32_t *len, uint8_t **data);
-static int usb_transport_set_report(struct usb_setup_packet *setup,
+static int usb_transport_set_report(const struct device *dev,
+		struct usb_setup_packet *setup,
 		int32_t *len, uint8_t **data);
-static int usb_transport_set_idle(struct usb_setup_packet *setup,
+static int usb_transport_set_idle(const struct device *dev,
+		struct usb_setup_packet *setup,
 		int32_t *len, uint8_t **data);
-static int usb_transport_set_protocol(struct usb_setup_packet *setup,
+static int usb_transport_set_protocol(const struct device *dev,
+		struct usb_setup_packet *setup,
 		int32_t *len, uint8_t **data);
-static void usb_transport_host_ready(void);
+static void usb_transport_host_ready(const struct device *dev);
 
 static const struct hid_ops usb_transport_callbacks = {
 	.get_report	= usb_transport_get_report,
@@ -155,28 +161,32 @@ int usb_transport_send_reply(uint8_t *data, uint32_t len)
 	return ret;
 }
 
-static int usb_transport_get_report(struct usb_setup_packet *setup,
+static int usb_transport_get_report(const struct device *dev,
+		struct usb_setup_packet *setup,
 		int32_t *len, uint8_t **data)
 {
 	LOG_DBG("usb_transport_get_report");
 	return 0;
 }
 
-static int usb_transport_get_idle(struct usb_setup_packet *setup,
+static int usb_transport_get_idle(const struct device *dev,
+		struct usb_setup_packet *setup,
 		int32_t *len, uint8_t **data)
 {
 	LOG_DBG("usb_transport_get_idle");
 	return 0;
 }
 
-static int usb_transport_get_protocol(struct usb_setup_packet *setup,
+static int usb_transport_get_protocol(const struct device *dev,
+		struct usb_setup_packet *setup,
 		int32_t *len, uint8_t **data)
 {
 	LOG_DBG("usb_transport_get_protocol");
 	return 0;
 }
 
-static int usb_transport_set_report(struct usb_setup_packet *setup,
+static int usb_transport_set_report(const struct device *dev,
+		struct usb_setup_packet *setup,
 		int32_t *len, uint8_t **data)
 {
 	uint8_t	*buffer;
@@ -190,21 +200,23 @@ static int usb_transport_set_report(struct usb_setup_packet *setup,
 	return 0;
 }
 
-static int usb_transport_set_idle(struct usb_setup_packet *setup,
+static int usb_transport_set_idle(const struct device *dev,
+		struct usb_setup_packet *setup,
 		int32_t *len, uint8_t **data)
 {
 	LOG_DBG("usb_transport_set_idle");
 	return 0;
 }
 
-static int usb_transport_set_protocol(struct usb_setup_packet *setup,
+static int usb_transport_set_protocol(const struct device *dev,
+		struct usb_setup_packet *setup,
 		int32_t *len, uint8_t **data)
 {
 	LOG_DBG("usb_transport_set_protocol");
 	return 0;
 }
 
-static void usb_transport_host_ready(void)
+static void usb_transport_host_ready(const struct device *dev)
 {
 	LOG_DBG("usb_transport_host_ready");
 }

